src/Symfony/Component/Console/Output/StreamOutput.php

```diff
@@ -106,10 +107,22 @@ protected function hasColorSupport()
             return true;
         }
 
-        return 'Hyper' === getenv('TERM_PROGRAM')
+        if ('Hyper' === getenv('TERM_PROGRAM')
+            || false !== getenv('COLORTERM')
             || false !== getenv('ANSICON')
             || 'ON' === getenv('ConEmuANSI')
-            || str_starts_with((string) getenv('TERM'), 'xterm');
+        ) {
+            return true;
+        }
+
+        $term = (string) getenv('TERM');
+
+        if ('dumb' === $term) {
+            return false;
+        }
+
+        // See https://github.com/chalk/supports-color/blob/d4f413efaf8da045c5ab440ed418ef02dbb28bf1/index.js#L157
+        return 1 === @preg_match('/^((xterm|screen|putty|rxvt|linux).*)|(.*-256color(-bce)?)$/', $term);
```

Comment:
why not taking the full list of prefixes from the chalk code ?

And for the suffix, you support `-256color` but not `-256`

Reply:
The handling is a bit different for windows (and I don't have a windows machine so I can't easily test it); so I do not know if `vt100` and co. should be in there.

That said it is missing `ansi` at the very least.